app/controllers/foreman_deployments/api/v2/deployments_controller.rb

```diff
@@ -0,0 +1,73 @@
+module ForemanDeployments
+  module Api
+    module V2
+      class DeploymentsController < BaseController
+        before_filter :find_resource, :only => [:show, :configure, :configuration, :run]
+        before_filter :parse_stack, :only => [:show, :configure, :configuration, :run]
+
+        def_param_group :deployment do
+          param :deployment, Hash, :required => true, :action_aware => true do
+            param :name, String, :required => true, :desc => N_('Name for the deployment')
+            param :stack_id, :identifier, :required => true, :desc => N_('Id of the stack to deploy')
+          end
+        end
+
+        api :POST, '/deployments/', N_('Create a deployment')
+        param_group :deployment, :as => :create
+        def create
+          deployment_params = params[:deployment]
+
+          stack = ForemanDeployments::Stack.find(deployment_params.delete(:stack_id)) if deployment_params[:stack_id]
+          deployment_params[:configuration] = ForemanDeployments::Configuration.new(:stack => stack)
```

Comment:
Shouldn't we check for saved config before defining a new one for the deployment?